apps/files/js/newfilemenu.js

```diff
@@ -224,10 +234,11 @@
 		 * Renders the menu with the currently set items
 		 */
 		render: function() {
+			const menuItems = this._menuItems.filter(item => !item.shouldShow || (item.shouldShow instanceof Function && item.shouldShow() === true))
```

Comment:
If i understand this correctly you want it to show if `item.shouldShow` is not set or it's a function and the function returns true.

If `shouldShow` is false this will still show. If `shouldShow` is true it will not show, right?
I think that's confusing and might cause errors.

Instead i suggest:
```suggestion
			const menuItems = this._menuItems.filter(item => !item.shouldShow?.() ?? true)
```

MDN has docs for the [? operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) and the [?? operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing).

Reply:
@max-nextcloud The filter function only returns the entries for which the conditions set are true.
So only the entries for which `shouldShow` is not defined and those for which `shouldShow` is a function that return true will be displayed.

Reply:
@luka-nextcloud - what happens if shouldShow is `false` or `true` ?

Reply:
I'd say the more explicit check if shouldShow is a function makes sense as otherwise passing true or false as values . 

`shouldShow?.()` still seems to trigger an error in that case:

![Screenshot 2023-01-25 at 15 36 25](https://user-images.githubusercontent.com/3404133/214591518-fd69e145-a601-41b6-8581-46c76bfa24ea.png)

Reply:
@max-nextcloud 
First of all, `shouldShow` can not be a bool, it should be a function.
If shouldShow is false, the entry won't be shown.
If shouldShow returns true as a function then it'll be shown.

Reply:
> `shouldShow?.()` still seems to trigger an error in that case:

Interesting `?.()` seems to only work with undefined - not with false. I learned something today :)